SOD.Common/Patches/InputControllerPatches.cs

```diff
@@ -48,19 +48,22 @@ internal static void Prefix(InputController __instance)
                         _gameMappedKeyDictionary.Add(capitalizedKey, Enum.Parse<InteractablePreset.InteractionKey>(key, true));
                     }
                 }
+                if (!__instance.player.GetAnyButtonDown() && !__instance.player.GetAnyButtonUp())
+                {
+                    return;
+                }
                 foreach (var actionName in _actionNames)
                 {
-                    var key = _gameMappedKeyDictionary.ContainsKey(actionName) ? _gameMappedKeyDictionary[actionName] : InteractablePreset.InteractionKey.none;
-                    if (__instance.player.GetButtonDown(actionName))
-                    {
-                        Lib.InputDetection.ReportButtonStateChange(actionName, key, true);
-                    }
-                    else if (__instance.player.GetButtonUp(actionName))
+                    if (__instance.player.GetButtonDown(actionName) || __instance.player.GetButtonUp(actionName))
                     {
-                        Lib.InputDetection.ReportButtonStateChange(actionName, key, false);
+                        var key = _gameMappedKeyDictionary.ContainsKey(actionName)
+                            ? _gameMappedKeyDictionary[actionName]
+                            : InteractablePreset.InteractionKey.none;
+        
+                        Lib.InputDetection.ReportButtonStateChange(actionName, key, __instance.player.GetButtonDown(actionName));
```

Comment:
Possibly assign the result of `__instance.player.GetButtonDown(actionName)` to a variable and reuse it across lines 57 and 63. I'm not sure what Rewired does under-the-hood for these calls ([API](https://guavaman.com/projects/rewired/docs/api-reference/html/M_Rewired_Player_GetButtonDown_1.htm)).

Reply:
Good catch! Committed the change.